src/Jackalope/Node.php

```diff
@@ -1608,7 +1633,8 @@ protected function _setProperty($name, $value, $type, $internal)
     public function setDirty($keepChanges = false, $targetState = false)
     {
         parent::setDirty($keepChanges, $targetState);
-        foreach ($this->properties as $property) {
+        foreach ($this->propertyValues as $name => $propertyValue) {
+            $property = $this->loadProperty($name);
```

Comment:
we only need to do this for already loaded properties. the ones not created yet do not need to be re-created later.